### ISSUE TITLE
Is this banner crash a MoPub issue or an issue with the custom adapter?

### ISSUE THREAD

**Issue by casolorz** [criteo/android-publisher-sdk#246]
I have submitted this to MoPub but they told me to submit it here https://github.com/mopub/mopub-android-sdk/issues/511

MoPub SDK Version: 5.18.0
Criteo SDK 4.3.0

I've only seen this crash on Crashlytics so I don't have any more details and it hasn't happened to me on testing even when testing that ad network.

```
Fatal Exception: java.lang.IllegalStateException: Calling View methods on another thread than the UI thread.
       at com.android.webview.chromium.WebViewChromium.b(WebViewChromium.java:2)
       at com.android.webview.chromium.WebViewChromium.init(WebViewChromium.java:15)
       at android.webkit.WebView.<init>(WebView.java:648)
       at android.webkit.WebView.<init>(WebView.java:577)
       at android.webkit.WebView.<init>(WebView.java:560)
       at android.webkit.WebView.<init>(WebView.java:547)
       at android.webkit.WebView.<init>(WebView.java:537)
       at com.criteo.publisher.CriteoBannerView.<init>(CriteoBannerView.java:127)
       at com.criteo.publisher.CriteoBannerView.<init>(CriteoBannerView.java:118)
       at com.criteo.mediation.mopub.CriteoBannerAdapter.load(CriteoBannerAdapter.java)
       at com.mopub.mobileads.BaseAd.internalLoad(BaseAd.java:104)
       at com.mopub.mobileads.AdAdapter.load(AdAdapter.java:84)
       at com.mopub.mobileads.AdViewController.loadBaseAd(AdViewController.java:845)
       at com.mopub.mobileads.AdViewController$3.onSettingsReceived(AdViewController.java:219)
       at com.mopub.common.CESettingsCacheService$getCESettings$cacheListener$1.onGetComplete(CESettingsCacheService.java:101)
       at com.mopub.common.CacheService$getFromDiskCacheAsync$$inlined$CoroutineExceptionHandler$1.handleException(CacheService.java:113)
       at kotlinx.coroutines.CoroutineExceptionHandlerKt.handleCoroutineException(CoroutineExceptionHandlerKt.java:25)
       at kotlinx.coroutines.StandaloneCoroutine.handleJobException(StandaloneCoroutine.java:193)
       at kotlinx.coroutines.JobSupport.finalizeFinishingState(JobSupport.java:229)
       at kotlinx.coroutines.JobSupport.tryMakeCompletingSlowPath(JobSupport.java:906)
       at kotlinx.coroutines.JobSupport.tryMakeCompleting(JobSupport.java:863)
       at kotlinx.coroutines.JobSupport.makeCompletingOnce$kotlinx_coroutines_core(JobSupport.java:828)
       at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.java:111)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(BaseContinuationImpl.java:46)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.java:104)
       at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.java:571)
       at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.java:750)
       at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.java:678)
       at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.java:665)
```

**Comment by TurpIF** [comment]
Hi,

Thank you for raising this issue. Currently I believe that the issue is on MoPub side. So to ease the discussion, I'll continue on the MoPub's thread until we defined from where, Criteo or MoPub, it comes from.

**Comment by TurpIF** [comment]
Hi @casolorz ,

I opened a PR to handle the issue. Still, I suggest you to try to call MoPub on UI thread.
As we were preparing a 4.6.0.0 version, the fix will be available on this next version.

**Comment by casolorz** [comment]
Thanks for handling this. Just fyi I am calling everything on the UI thread, something else is putting it on a worker thread, I'm guessing it is the CoroutineExceptionHandler.

**Comment by TurpIF** [comment]
Hi,

I'm closing this issue as the fixed has been deployed on the adapter 4.6.0.0: https://github.com/criteo/android-publisher-sdk-mopub-adapters/releases/tag/v4.6.0.0.

Do not hesitate to reopen it if you have any trouble with the fix, and good luck with the CoroutineExceptionHandler issue.
